### PR TITLE
Add Permit2 Mint/Redeem for Chambers

### LINKED ISSUE #11: :sparkles: Add Mint/Redeem for Chambers

### PR DESCRIPTION
**Pull request by Gaonuk**
Closes #11

### CODE REVIEW

File: test/permit2/SetTokens/mintSetWithPermit2.t.sol [hunk 1]
```diff
@@ -22,17 +22,18 @@ contract GaslessTest is Test, Permit2Utils, DeployPermit2 {
     using SafeTransferLib for ERC20;
     using SafeTransferLib for ISetToken;
 
-    bytes32 constant FULL_EXAMPLE_WITNESS_TYPEHASH = keccak256(
+    bytes32 constant WITNESS_TYPEHASH = keccak256(
         "PermitWitnessTransferFrom(TokenPermissions permitted,address spender,uint256 nonce,uint256 deadline,MintData witness)MintData(ISetToken _setToken,uint256 _amountSetToken,uint256 _maxAmountInputToken, bytes[] _componentQuotes,address _issuanceModule,bool _isDebtIssuance)TokenPermissions(address token,uint256 amount)"
     );
 
-    bytes32 public constant _TOKEN_PERMISSIONS_TYPEHASH = keccak256("TokenPermissions(address token,uint256 amount)");
+    bytes32 public constant _TOKEN_PERMISSIONS_TYPEHASH =
```
Comment (JPSCHELE) [inline test/permit2/SetTokens/mintSetWithPermit2.t.sol:29]: Why the underscore prefix ?

File: src/Gasworks.sol [hunk 2]
```diff
@@ -82,6 +94,19 @@ contract Gasworks is ERC2771Recipient, Owned {
...
+        IIssuerWizard _issuerWizard;
+        // The address of the token used to mint
+        IERC20 _baseToken;
+        // Maximum amount of baseToken to use to mint
+        uint256 _maxPayAmount;
+        // The amount of Chamber to mint
+        uint256 _mintAmount;
+    }
+
     struct RedeemData {
```
Comment (JPSCHELE) [inline src/Gasworks.sol:110]: Here AW, RedeemTokenSetData ?

File: src/Gasworks.sol [hunk 3]
```diff
@@ -27,6 +33,12 @@ contract Gasworks is ERC2771Recipient, Owned {
     string private constant REDEEMDATA_WITNESS_TYPE_STRING =
```
[the remark this answers is not in the document]
Comment (Gaonuk) [reply src/Gasworks.sol:33]: Makes sense, will do

File: test/permit2/SetTokens/mintSetWithPermit2.t.sol (hunk 1 above)
Comment (Gaonuk) [reply test/permit2/SetTokens/mintSetWithPermit2.t.sol:29]: Copy pasted the value from Permit2 repo and that was the name of the variable, will remove the underscore to keep consistency

File: src/interfaces/IWETH.sol
[the remark this answers is not in the document]
Comment (Gaonuk) [reply src/interfaces/IWETH.sol:1]: Actually i don't know lol

File: src/Gasworks.sol [hunk 4]
```diff
@@ -442,57 +439,64 @@ contract Gasworks is ERC2771Recipient, Owned {
...
+        if (balance == 0) revert ZeroBalance(address(token));
+        token.safeTransfer(owner, balance);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * Performs a low-lever call to swapTarget to perform a swap between two tokens
```
Comment (JPSCHELE) [inline src/Gasworks.sol:472]:
```suggestion
     * Performs a low-level call to swapTarget to perform a swap between two tokens
```

**Review by JPSCHELE — APPROVED** [review]
Small typo in a Docstring

**Review by JPSCHELE — APPROVED** [review]

**Pull request merged**